### PR TITLE
CodeCamp #38 RTMDet-R

### PR DESCRIPTION
**Pull request by liuyanyi**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmrotate/models/losses/gaussian_dist_loss.py [hunk 1]
```diff
@@ -386,7 +386,9 @@ def forward(self,
             reduction_override if reduction_override else self.reduction)
         if (weight is not None) and (not torch.any(weight > 0)) and (
                 reduction != 'none'):
-            return (pred * weight).sum()
+            if pred.dim() == weight.dim() + 1:
```
Comment (zytx121) [inline mmrotate/models/losses/gaussian_dist_loss.py:389]: Why do we need to change this file?

File: configs/rotated_rtmdet/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+| :------: | :------: | :------: | :-------------: | :---: | :---: | :---: | :-------------------------------------------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+| RTMDet-l |    IN    |   100e   | Mixup+Mosaic+RR | 54.59 | 80.16 | 61.16 | [config](./rotated_rtmdet_l-100e-aug-dota.py) | [model](https://download.openmmlab.com/mmrotate/v1.0/rotated_rtmdet/rotated_rtmdet_l-100e-aug-dota/rotated_rtmdet_l-100e-aug-dota-bc59fd88.pth) \| [log](https://download.openmmlab.com/mmrotate/v1.0/rotated_rtmdet/rotated_rtmdet_l-100e-aug-dota/rotated_rtmdet_l-100e-aug-dota_20221124_224135.json) |
+
+**Note**:
+
+1. We follow the latest metrics from the DOTA evaluation server, original voc format mAP is now mAP50.
+2. `IN` means ImageNet pretrain, `COCO` means COCO pretrain.
+3. Different from the report, the inference speed here is measured on an NVIDIA 2080Ti GPU with TensorRT 8.4.3, cuDNN 8.2.0, FP16, batch size=1, and with NMS.
```
Comment (zytx121) [inline configs/rotated_rtmdet/README.md:63]:
```
@misc{lyu2022rtmdet,
      title={RTMDet: An Empirical Study of Designing Real-Time Object Detectors}, 
      author={Chengqi Lyu and Wenwei Zhang and Haian Huang and Yue Zhou and Yudong Wang and Yanyi Liu and Shilong Zhang and Kai Chen},
      year={2022},
      eprint={2212.07784},
      archivePrefix={arXiv},
      primaryClass={cs.CV}
}
```

File: configs/rotated_rtmdet/rotated_rtmdet_l-100e-aug-dota.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+# learning rate
+param_scheduler = [
+    dict(
+        type='LinearLR',
+        start_factor=1.0e-5,
+        by_epoch=False,
+        begin=0,
+        end=1000),
+    dict(
+        # use cosine lr from 150 to 300 epoch
```
Comment (zytx121) [inline configs/rotated_rtmdet/rotated_rtmdet_l-100e-aug-dota.py:163]:
```suggestion
        # use cosine lr from 50 to 100 epoch
```

File: configs/rotated_rtmdet/rotated_rtmdet_l-9x-hrsc.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+# learning rate
+param_scheduler = [
+    dict(
+        type='LinearLR',
+        start_factor=1.0e-5,
+        by_epoch=False,
+        begin=0,
+        end=1000),
+    dict(
+        # use cosine lr from 150 to 300 epoch
```
Comment (zytx121) [inline configs/rotated_rtmdet/rotated_rtmdet_l-9x-hrsc.py:94]:
```suggestion
        # use cosine lr from 54 to 108 epoch
```

File: mmrotate/models/losses/gaussian_dist_loss.py (hunk 1 above)
Comment (liuyanyi) [reply mmrotate/models/losses/gaussian_dist_loss.py:390]: GDLoss got error when used with RTMDet-R.
Follow other losses in mmdet, handle the different input dim of weight.

**Review by zytx121 — APPROVED** [review]
LGTM

**Review by RangiLyu — APPROVED** [review]

**Pull request merged**